sources/api/migration/migrator/src/main.rs

```diff
@@ -323,6 +345,22 @@ fn remove_metadata_from_datastore(datastore: &mut DataStoreData) -> Result<()> {
     Ok(())
 }
 
+// Removes all data from the datastore where the key begins with the given prefix
+fn remove_keys_with_prefix(datastore: &mut DataStoreData, prefix: &str) -> Result<()> {
+    let keys_to_remove = datastore
+        .data
+        .keys()
+        .filter(|key| key.starts_with(prefix))
+        .cloned()
```

Comment:
Open question - is it possible to avoid the clone here and perform the removal in the one-pass? Not sure how costly the copy would be. If performance is not a concern, I am good with the change as the readability is great.

Reply:
In general, not without writing unsafe code because the first iterator borrows the key immutably, which in turn borrows the datastore. Since that borrow is held, we can't simultaneously perform a mutable borrow to perform the deletion.

But! The HashMap implementation actually has the requisite `unsafe` code [abstracted away in the `retain` method](https://github.com/rust-lang/hashbrown/blob/master/src/map.rs#L917). Thanks for prompting me to look!